salt/modules/file.py

```diff
@@ -1651,8 +1651,13 @@ def replace(path,
         with salt.utils.fopen(path,
                               mode='r',
                               buffering=bufsize) as r_file:
+            count_replaced = 0
             for line in r_file:
-                result, nrepl = re.subn(cpattern, repl, line, count)
+                if count == 0 or count_replaced != count:
```

Comment:
I see that, but that's exactly what causes `count_replaced` to become greater than `count` with multiple occurrences. 

Say, `count=5`, there's 1 occurrence on 4 lines, then 2 occurrences on one line, then 1 occurrence on the next 4 lines. This code will get to `count_replaced=4`, then jump to `count_replaced=6`, and since `6 != 5` it will keep going to replace all 10 occurrences. No? What am I not seeing?

If that's correct, even changing this test to `count_replaced < count` won't quite work entirely as it would replace 6 occurrences in that example... the only solution I can think of would be to decrement count (`count -= nrepl`) and break if `count<=0`, with a flag to skip this bit if `count=0` from the start.